@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -462,38 +488,21 @@ export class TopViewClipPlanesEvaluator extends ElevationBasedClipPlanesEvaluato
         // There may be situations when maximum elevation still remains below sea level
         // (elevation < 0) or it is negligible (elevation ~ epsilon), in such cases tangent
         // extension (te) is not required.
-        const te =
-            this.maxElevation < epsilon
-                ? 0
-                : Math.sqrt(2 * r * this.maxElevation + this.maxElevation * this.maxElevation);
+        const te = this.getTangentDistance(r + this.maxElevation, r);
 
         // Next step is to project CT vector onto camera eye (forward) vector to get maximum
         // camera far plane distance. Point on sphere beyond that point won't be visible
         // anyway unless they are above the ground surface.
         // For such cases [[this.maxElevation]] applies, thus we project entire CE
         // vector of length equal: t + te.
 
-        // Extract camera X, Y, Z orientation axes into tmp vectors array.
-        camera.matrixWorld.extractBasis(
-            this.m_tmpVectors[0],
-            this.m_tmpVectors[1],
-            this.m_tmpVectors[2]
-        );
-        // Setup quaternion (based on X axis) for angle between tangent and camera eye.
-        this.m_tmpQuaternion.setFromAxisAngle(this.m_tmpVectors[0], alpha);
-        // Acquire camera (eye) forward vector from Z axis reversed (keep in tmpVectors[2]).
-        const cameraFwdVec = this.m_tmpVectors[2].negate();
-        // Apply quaternion to forward vector, creating tangent vector (in tmpVectors[1]).
-        const tangentVec = this.m_tmpVectors[1]
-            .copy(cameraFwdVec)
-            .applyQuaternion(this.m_tmpQuaternion);
-        // Give it a proper length
-        tangentVec.multiplyScalar(t + te);
-        // Calculate tangent projection onto camera eye vector, giving far plane distance.
-        return tangentVec.dot(cameraFwdVec);
+        // Calculate tangent projection onto camera eye vector, giving far plane distance
```

Comment:
I know this seems like a huge change but the code is actually equivalent.
We know that `tangentVec.dot(cameraFwdVec) = Math.cos(alpha) * tangentVec.length() * cameraFwdVec.length()`.
Because `cameraFwdVec.length() = 1` and ` tangentVec.length() == t + te` the formula can be simplified to ` Math.cos(alpha) * (t + te)`.